.github/workflows/deploy.yml

```diff
@@ -87,7 +88,7 @@ jobs:
       - uses: actions/upload-artifact@v3
         with:
           path: ./wheelhouse/*.whl
-          name: publish-shared-wheels
+          name: aer-deploy-wheel-arm64-macos-${{ matrix.os }}
```

Comment:
This is still `upload-artifact@v3` and needs bumping or the `download-artifact@v4` won't be able to find it.

This is the only other `upload-artifact` run (along with the top comment) that gets downloaded by the shared uploader - all the others below it in this file might not want to match the prefix exactly.

Reply:
Bumped to v4 in d9d7fb78 and changed the name to include "shared". I was hesitant to change to v4 in my original PR since I am not aware of the reasons to the original bump to v4 and why only it affected only part of the file.

Reply:
The change from v3 to v4 was a huge breaking change in the upload/download-artifact jobs.  You certainly always have to match (v4 download can't find v3 uploads, and so on), and I think GitHub will be pushing to deprecate and remove the old v3 system soon too - iirc, it was completely piggy-backed onto their logging framework, and never properly designed as an artifact store.